src/MoBi.Core/Services/SimulationConfigurationFactory.cs

```diff
@@ -0,0 +1,40 @@
+﻿using MoBi.Core.Domain.Model;
+using OSPSuite.Core.Domain.Builder;
+using OSPSuite.Core.Domain.Formulas;
+using OSPSuite.Core.Domain.Services;
+using OSPSuite.Utility.Extensions;
+
+namespace MoBi.Core.Services
+{
+   public interface ISimulationConfigurationFactory
+   {
+      SimulationConfiguration Create();
+   }
+   
+   public class SimulationConfigurationFactory : ISimulationConfigurationFactory
+   {
+      private readonly ICoreCalculationMethodRepository _calculationMethodRepository;
+      private readonly ICloneManagerForBuildingBlock _cloneManager;
+      private readonly IMoBiContext _context;
+
+      public SimulationConfigurationFactory(ICoreCalculationMethodRepository calculationMethodRepository, ICloneManagerForBuildingBlock cloneManager, IMoBiContext context)
+      {
+         _calculationMethodRepository = calculationMethodRepository;
+         _cloneManager = cloneManager;
+         _context = context;
+      }
+
+      public SimulationConfiguration Create()
+      {
+         var simulationConfiguration = new SimulationConfiguration
+         {
+            SimulationSettings = _cloneManager.Clone(_context.CurrentProject.SimulationSettings)
+         };
+
+         _calculationMethodRepository.All().Each(
+            cm => simulationConfiguration.AddCalculationMethod(_cloneManager.Clone(cm, new FormulaCache())));
```

Comment:
what's with the new formula cache here? This looks suspicious. Aren't they also buildingBlocks?

Reply:
That's a copy/paste from BuildConfigurationFactory in develop branch. I wonder if that was a compensation that isn't needed now. I'll try it and update if it works.